catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy

```diff
@@ -0,0 +1,283 @@
+
+        String zip = tmpHomeDir.listFiles().find()?.canonicalPath
+        assert zip?.trim() as boolean // null or empty
+        assert zip.endsWith('.zip')
+
+        def files = new ZipFile(zip).entries()
+                .collect { it.isDirectory() ? null : it.name }
+                .findAll { it != null }
+
+        assert [result.metacard.id].every { id ->
+            files.any { it.contains(id) }
+        }
+
+
+    }
+
+    def "Test single metacard with content export"() {
+        setup:
+        exportCommand.with {
+            it.delete = false
+        }
+
+        def result = new ResultImpl(simpleMetacard(simpleAttributes() + [(Metacard.TAGS): [Metacard.DEFAULT_TAG]]))
+        def resourceName = "contentfor-${result.metacard.id}.xml" as String
+        exportCommand.catalogFramework = Mock(CatalogFramework) {
+            query(_ as QueryRequest) >> { QueryRequest req ->
+                new QueryResponseImpl(req, [result], 1)
+            }
+            getLocalResource(_ as ResourceRequest) >> { ResourceRequest req ->
+                BinaryContent xmlContent = getMetacardToXmlTransformer()(result.metacard, [:])
+
+                return new ResourceResponseImpl(req, [:], new ResourceImpl(xmlContent.inputStream,
+                        new MimeType('text/xml'),
+                        resourceName))
+            }
+        }
+
+        when:
+        exportCommand.executeWithSubject()
+
+        then:
+        notThrown(Exception)
+        tmpHomeDir.list().size() == 1
+        tmpHomeDir.list().find()?.endsWith('.zip')
+
+        String zip = tmpHomeDir.listFiles().find()?.canonicalPath
+        assert zip?.trim() as boolean // null or empty
+        assert zip.endsWith('.zip')
+
+        def files = new ZipFile(zip).entries()
+                .collect { it.isDirectory() ? null : it.name }
+                .findAll { it != null }
+        println(files)
+        assert [result.metacard.id].every { id ->
+            files.any { it.contains(id) }
+        }
+        assert [resourceName].every { name ->
+            files.any { it.contains(name) }
+        }
+
+    }
+
+/**************************************************************************
+ *
+ * Utility Methods
+ *
+ *************************************************************************/
+
+    Metacard simpleMetacard(Map kwargs) {
+        Metacard metacard = new MetacardImpl()
+        kwargs.forEach({ key, val ->
+            if (key != null && val != null) {
+                metacard.setAttribute(new AttributeImpl(key, val))
+            }
+        })
+        return metacard
+    }
+
+    Map simpleAttributes() {
+        String id = randomUUID()
+        [
+                (Metacard.ID)          : id,
+                (Metacard.TITLE)       : 'Metacard Title',
+                (Metacard.RESOURCE_URI): "content:$id".toString(),
+        ]
+    }
+
+    String randomUUID() {
+        UUID.randomUUID().toString().replaceAll('-', '')
```

Comment:
I just need a random ID. I prefer random over a hardcoded so that way you ensure just in case there's some weird state carryover between tests that you are using a new random ID so there will never be conflict.